setup.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -eu
+
+# Should this script error, print out the line that was responsible
+_error_report() {
+  echo >&2 "Exited [$?] at line $(caller):"
+  cat -n "$0" | tail -n+$(($1 - 3)) | head -n7 | sed "4s/^\s*/>>> /"
+}
+trap '_error_report $LINENO' ERR
+
+# Check if terraform.tfvars exists and read project name and region if available
+if [ -f terraform.tfvars ]; then
+  echo -e "\n\xe2\x88\xb4 Found terraform.tfvars. Store variables from the file. "
+  # Read project name and region from terraform.tfvars
+  TF_PROJECT=$(grep -Po '(?<=project_id = ")[^"]*' terraform.tfvars)
+  REGION=$(grep -Po '(?<=region = ")[^"]*' terraform.tfvars)
+else
+  # Check if there are arguments and they have the correct format
+  if [ $# -ne 5 ]; then
+    echo "Usage: $0 project_id=string region=string enable_apis=boolean force_destroy=boolean deletion_protection=boolean"
+    exit 1
+  fi
+
+  # Define an array of expected keys
+  EXPECTED_KEYS=("project_id" "region" "enable_apis" "force_destroy" "deletion_protection")
+
+  # Check if arguments have the correct format and keys
+  for arg in "$@"; do
+    IFS="=" read -r key value <<< "$arg"
+
+    found=false
+    for expected_key in "${EXPECTED_KEYS[@]}"; do
+      if [ "$key" = "$expected_key" ]; then
+        found=true
+        break
+      fi
+    done
+
+    if [ "$found" = false ] || ! [[ "$arg" =~ ^[a-zA-Z_][a-zA-Z0-9_]*=.*$ ]]; then
+      echo "Arguments must be in the format: key=value, and the key must be one of 'project_id', 'region', 'enable_apis', 'force_destroy', 'deletion_protection'."
```

Comment:
Replaced with a new variable, `KEYS_STR=$(IFS=", "; echo "${EXPECTED_KEYS[*]}")`. The space between comma didn't applied. It can be feasible by adding a few lines of code, but I think it is not necessary. So the output looks like the following:
`Arguments must be in the format: key=value, and the key must be one of: project_id,region,enable_apis,force_destroy.`